synapse/util/task_scheduler.py

```diff
@@ -94,10 +99,12 @@ def __init__(self, hs: "HomeServer"):
 
         if self._run_background_tasks:
             self._clock.looping_call(
-                run_as_background_process,
+                self._launch_scheduled_tasks,
+                TaskScheduler.SCHEDULE_INTERVAL_MS,
+            )
+            self._clock.looping_call(
+                self._clean_scheduled_tasks,
```

Comment:
I also separated in 2 like you did at some point, but I was not sure that no races exist between the 2 so I ended up removing it.
After more thinking I think we should be safe, we are iterating on 2 disjoint set of tasks (active+schedule vs failed+complete) and I don't see any trouble if an active task becomes complete or failed in the middle.